op-e2e/system_test.go

```diff
@@ -1352,11 +1364,25 @@ func testFees(t *testing.T, cfg SystemConfig) {
 	require.NoError(t, err)
 	require.Equal(t, sys.RollupConfig.IsEcotone(header.Time), gpoEcotone, "GPO and chain must have same ecotone view")
 
+	gpoFjord, err := gpoContract.IsFjord(nil)
+	require.NoError(t, err)
+	require.Equal(t, sys.RollupConfig.IsFjord(header.Time), gpoFjord, "GPO and chain must have same fjord view")
+
 	gpoL1Fee, err := gpoContract.GetL1Fee(&bind.CallOpts{}, bytes)
 	require.Nil(t, err)
 
 	adjustedGPOFee := gpoL1Fee
-	if sys.RollupConfig.IsRegolith(header.Time) {
+	if sys.RollupConfig.IsFjord(header.Time) {
+		// The fastlz size of the transaction is 102 bytes
+		require.Equal(t, uint64(102), tx.RollupCostData().FastLzSize)
+		// Which results in both the fjord cost function and GPO using the minimum value for the fastlz regression:
+		// Geth Linear Regression: -42.5856 + 102 * 0.8365 = 42.7374
+		// GPO Linear Regression: -42.5856 + 170 * 0.8365 = 99.6194
+		// The additional 68 (170 vs. 102) is due to the GPO adding 68 bytes to account for the signature.
```

Comment:
This case is similar to this [comment](https://github.com/ethereum-optimism/optimism/pull/9618#discussion_r1599863910). Were effectively passing the txn data (which is already signed) the GPO then adds 68 whereas op-geth does not.

The other way is to pass the unsigned data to the GPO, e.g.:

```
if regolith {
    bytes = bytes[:len(bytes)-68]
}
```

We'll need that if statement due to [pre-regolith cost func adding 68 non-zero bytes](https://github.com/ethereum-optimism/op-geth/blob/3704c9fa171e9688b9b54a7620d610a2b46d49a8/core/types/rollup_cost.go#L198-L202). This will allow us to remove the existing [edge case](https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/system_test.go#L1359-L1365
) too.

Though I suspect this will fail if there is ever signature that contains any 0 bytes.